recipe_scrapers/aldi.py

```diff
@@ -0,0 +1,97 @@
+# mypy: allow-untyped-defs
+import re
+
+from ._abstract import AbstractScraper
+from ._exceptions import ElementNotFoundInHtml
+from ._utils import get_minutes, get_yields
+
+
+class Aldi(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "aldi.com.au"
+
+    def site_name(self):
+        return "Aldi"
+
+    def author(self):
+        return self.soup.find("meta", {"name": "author"}).get("content")
+
+    def title(self):
+        return self.soup.find("h1").text
+
+    def category(self):
+        title = self.soup.select("a.tab-nav--link.dropdown--list--link.m-active")[
+            0
+        ].text
+        recipe_position = title.find(" Recipe")
+        return title[:recipe_position]
```

Comment:
```suggestion
        element = self.soup.select_one("a.tab-nav--link.dropdown--list--link.m-active")
        if element:
            title = element.text
            recipe_position = title.find(" Recipe")
            if recipe_position != -1:
                return title[:recipe_position]
```
replaced `soup.select` with `soup.select_one` to replace the list selection